src/Symfony/Bridge/PhpUnit/DeprecationErrorHandler/DeprecationNotice.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Symfony\Bridge\PhpUnit\DeprecationErrorHandler;
+
+final class DeprecationNotice
+{
+    /**
+     * @var int
+     */
+    private $count = 0;
+
+    /**
+     * @var array<string,int>
+     */
+    private $countsByCaller = [];
+
+    public function addObjectOccurence($class, $method)
+    {
+        if (!isset($this->countsByCaller["$class::$method"])) {
+            $this->countsByCaller["$class::$method"] = 0;
+        }
+        ++$this->countsByCaller["$class::$method"];
+        ++$this->count;
+    }
+
+    public function addProceduralOccurence()
+    {
+        ++$this->count;
+    }
+
+    public function countsByCaller()
```

Comment:
```suggestion
    public function getMethodOccurences()
```

Reply:
The method where the deprecation is triggered is not to be confused with the method that calls that method, and `countsByCallingMethod` seems a bit long, WDYT?

Reply:
I don't know, they seem symmetric to me so I would expect similar name. I guess we need 3rd opinion.

Reply:
`getCountsByCaller`?